test/e2e/validation_test.go

```diff
@@ -165,37 +155,50 @@ func validateInstanceSecret(kubeclient kubernetes.Interface, namespace, instance
 			return false, err
 		}
 		encodedCreds := instanceSecret.Data[instanceID]
-
-		var creds tracker.Credentials
 		if err := json.Unmarshal(encodedCreds, &creds); err != nil {
 			return false, errors.Wrap(err, "unmarshalling creds failed")
 		}
-
-		if err := validateConnectivity(ipAddress, instanceID, creds); err == nil {
-			log.Println("Successfully validated the SSH Connection")
-			return true, nil
-		}
-
-		log.Printf("failed with error for creds %v: %v", creds, err)
-		return false, nil
+		return true, nil
 	})
+	return creds, err
+}
+
+// validateInstanceSecret validates the instance secret.
+func (t *testContext) validateInstanceSecret(instanceID string) error {
```

Comment:
Please use a different receiver like `tc` for `testContext`. Otherwise it easy to mistake this for the `testing.T`. Applies to all methods associated with `testContext`.